src/main/java/com/ardaslegends/service/war/WarService.java

```diff
@@ -95,43 +99,94 @@ public War createWar(CreateWarDto createWarDto) {
         war = secureSave(war, warRepository);
 
+        discordService.sendMessageToRpChannel(WarMessages.declareWar(war, discordService));
+
         log.info("Successfully executed and saved new war {}", war.getName());
         return war;
     }
 
-    public Set<War> getWarsOfFaction(String factionName) {
+    public Set<War> getActiveWarsOfFaction(String factionName) {
         // TODO, not yet implemented
         return null;
     }
 
-    public Set<War> getWarsOfFaction(Faction faction) {
-        Set<War> wars = secureFind(faction, warRepository::findAllWarsWithFaction);
+    public Set<War> getActiveWarsOfFaction(Faction faction) {
+        Set<War> wars = secureFind(faction, warRepository::findAllActiveWarsWithFaction);
         return wars;
     }
 
-    private Role fetchFactionRole(Faction faction) {
-        Long roleId = faction.getFactionRoleId();
+    @Transactional(readOnly = false)
+    public War forceEndWar(EndWarDto dto) {
+        log.debug("Player with discord id [{}] is trying to force end war [{}]", dto.executorDiscordId(), dto.warName());
+
+        log.debug("Checking nulls and blanks");
+        ServiceUtils.checkAllNulls(dto);
+        ServiceUtils.checkAllBlanks(dto);
+
+        log.trace("Fetching player with discord id [{}]", dto.executorDiscordId());
+        val foundPlayer = playerRepository.findByDiscordID(dto.executorDiscordId());
+
+        if(foundPlayer.isEmpty()) {
+            log.warn("Player with id [{}] could not be found", dto.executorDiscordId());
+            throw PlayerServiceException.noPlayerFound(dto.executorDiscordId());
+        }
+
+        val player = foundPlayer.get();
+        log.debug("DiscordId [{}] belongs to player [{}]", dto.executorDiscordId(), player.getIgn());
+
+        log.debug("Player [{}] is staff: {}", player.getIgn(), player.getIsStaff());
+        if(!player.getIsStaff()) {
+            log.warn("Player [{}] is not a staff member and does not have the permission to force end wars!", player.getIgn());
+            throw StaffPermissionException.noStaffPermission();
+        }
+
+        val war = getActiveWarByName(dto.warName());
+        log.debug("Found war with name [{}]", war.getName());
+
+        log.debug("Ending war");
+        war.end();
+
+        discordService.sendMessageToRpChannel(WarMessages.forceEndWar(war, player, discordService));
+
+        log.info("War [{}] between attacker [{}] and defender [{}] has succesfully been ended by staff member [{}]", war.getName(), war.getInitialAttacker().getName(), war.getInitialDefender().getName(), player.getIgn());
+        return war;
+    }
+
+    public War getWarByName(String name) {
+        log.debug("Getting war with name [{}]", name);
+
+        Objects.requireNonNull(name, "War name must not be null!");
+        ServiceUtils.checkBlankString(name, "name");
+
+        log.debug("Fetching war with name [{}]", name);
+        val foundWar = warRepository.findByName(name);
```

Comment:
I think we're still using secureFind for spring generated reposiotry methods, right?

Reply:
Yes